astropy/units/core.py

```diff
@@ -1942,7 +1962,9 @@ def represents(self):
         """
         return self
 
-    def decompose(self, bases=set()):
+    def decompose(self, bases=None):
+        if bases is None:
+            bases = set()
```

Comment:
This is not needed, one would just do `if bases is not None and self not in bases:` below.